kolibri/core/public/test/test_api.py

```diff
@@ -461,6 +461,33 @@ def test_create_full_queue_should_queue(self):
             SyncQueue.objects.filter(id=queue_id, user_id=self.learner.id).exists()
         )
 
+    def test_create_full_queue_user_already_queued_should_resume(self):
+        for i in range(0, MAX_CONCURRENT_SYNCS):
+            learner = FacilityUser.objects.create(
+                username="test{}".format(i),
+                password="***",
+                facility=self.facility,
+            )
+            SyncQueue.objects.create(
+                user_id=learner.id,
+                keep_alive=10,
+            )
+        queue = SyncQueue.objects.create(user_id=self.learner.id, keep_alive=10)
+        time.sleep(1)
```

Comment:
have I ever said how much I hate adding sleep to the tests?... I know this is a bad windows behaviour so one more reason not-to-love that OS
btw, this discussion is related https://stackoverflow.com/questions/85451/pythons-time-clock-vs-time-time-accuracy

Reply:
Yeah, I am not a huge fan, but it seems like the only way :/